cfpb/consumerfinance.gov
# Deprecate filterable topic ordering by frequency

Currently when adding a filterable list to a page users can configure if or how they want topic filtering to appear. They can hide topic filtering entirely, have filterable topics be ordered alphabetically, or have them be filtered by their frequency of use (popularity).

As discussed on internal D&CP#220, we are deprecating the ability to sort by frequency, so that if topics appear, they are always sorted alphabetically. This change makes that happen.

This change includes a data migration that migrates existing StreamFields with a FilterableList block from the "topic_filtering" ChoiceBlock (which had 3 options: no_filter, sort_alphabetically, sort_by_frequency) to a simpler "topics" BooleanBlock, which determines whether topic sorting is enabled.

## How to test this PR

To test this change, verify that existing pages with a topic filter (like http://localhost:8000/about-us/blog/) continue to have one, and that topics are sorted alphabetically, and existing pages without a filter (like http://localhost:8000/compliance/circulars/) no longer do.

## Screenshots

|Before|After|
|-|-|
|<img width="480" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/0389a16e-c3c2-4fac-b0ab-82fd8488bcde"><img width="475" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/6c2104e3-1621-4f69-928f-ec9251daff6b">|<img width="407" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/ac068140-3f86-48b9-8d12-424c59cc7fa1">|

## Notes and todos

This PR includes a commit that fixes our Wagtail StreamField block data migration code (which we document [here](https://cfpb.github.io/consumerfinance.gov/migrations/#wagtail-specific-considerations)) so that it works properly on page revisions in Wagtail 4+. I was surprised that our unit tests didn't catch that failure, but it turns out the code will work under unit testing conditions but not under real migration conditions. It would be better if we could migrate to Wagtail's [StreamField data migration functionality](https://docs.wagtail.org/en/stable/advanced_topics/streamfield_migrations.html#streamfield-data-migrations) but we need to be on Wagtail 4.2. for that.

## Checklist
